src/module/item/sheet/rule-elements/base.ts

```diff
@@ -214,7 +253,12 @@ function cleanDataUsingSchema(schema: Record<string, DataField>, data: Record<st
         }
 
         if ("fields" in field) {
-            cleanDataUsingSchema(field.fields as Record<string, DataField>, data);
+            const maybeObject = data[key];
+            if (R.isObject(maybeObject)) {
+                cleanDataUsingSchema(field.fields as Record<string, DataField>, maybeObject);
+            } else {
+                cleanDataUsingSchema(field.fields as Record<string, DataField>, data);
+            }
```

Comment:
Shouldn't all of this act on data[key]? This was supposed to be the recursive case, and I believe acting on data was probably incorrect.

Reply:
I've added that check because it was passing non-objects in some instances.

Reply:
In those cases, what was the schema being passed?

Reply:
The `text` field on a `RollNoteRuleElement` which is a `DataUnionField` that can be a string or an object (I think).